src/useSpeech/index.ts

```diff
@@ -0,0 +1,89 @@
+import { useMemo, useState } from 'react';
+import { useDeepCompareEffect } from '../useDeepCompareEffect';
+
+type SpeechOptions = {
+	voice?: SpeechSynthesisVoice;
+	rate?: number;
+	pitch?: number;
+	volume?: number;
+	lang?: string;
+};
+
+type SpeechPlayStatus = 'init' | 'play' | 'pause' | 'end' | 'error';
+
+export type SpeechState = {
+	isPlaying: boolean;
+	status: SpeechPlayStatus;
+	errorMessage?: string;
+};
+
+function getInitState(): SpeechState {
+	return {
+		isPlaying: false,
+		status: 'init',
+	};
+}
+
+/**
+ * Hook for playing text using the SpeechSynthesis API.
+ *
+ * @param {string} text - Text to play.
+ * @param {SpeechOptions} options - Options for playing the text.
+ * @param {SpeechSynthesisVoice} [options.voice] - Voice to use for playing the text.
+ * @param {number} [options.rate] - Rate at which to play the text.
+ * @param {number} [options.pitch] - Pitch at which to play the text.
+ * @param {number} [options.volume] - Volume at which to play the text.
+ * @param {string} [options.lang] - Language of the text.
+ * @returns {SpeechState} State object representing the current state of speech synthesis.
+ */
+export const useSpeech = (text: string, options: SpeechOptions): SpeechState => {
+	const [state, setState] = useState<SpeechState>(getInitState);
+
+	const handleActions = useMemo(
+		() => ({
+			play() {
+				setState((preState) => {
+					return { ...preState, isPlaying: true, status: 'play' };
+				});
+			},
+			pause() {
+				setState((preState) => {
+					return { ...preState, isPlaying: false, status: 'pause' };
+				});
+			},
+			end() {
+				setState((preState) => {
+					return { ...preState, isPlaying: false, status: 'end' };
+				});
+			},
+			error(error: SpeechSynthesisErrorEvent) {
+				setState((preState) => {
+					return { ...preState, isPlaying: false, status: 'error', errorMessage: error.error };
+				});
+			},
+		}),
+		[]
+	);
```

Comment:
As said before - there is no need in creating these handlers within hook body - you can do it within body of effect function.

Reply:
@xobotyi Addressed [here](https://github.com/react-hookz/web/pull/1378/commits/978907134ccebce3f1d43317edd51a17060b8c82)